mempool/src/shared_mempool.rs

```diff
@@ -259,6 +269,158 @@ async fn process_incoming_transactions<V>(
     notify_subscribers(SharedMempoolNotification::NewTransactions, &smp.subscribers);
 }
 
+/// send callback here
+async fn process_rpc_submit_transactions_request<V>(
+    smp: SharedMempool<V>,
+    peer_id: PeerId,
+    request: SubmitTransactionsRequest,
+    callback: oneshot::Sender<Result<Bytes, RpcError>>,
+) where
+    V: TransactionValidation,
+{
+    // the RPC response object that will be returned to sender via callback
+    let mut response_msg = None;
+
+    if let Ok(response) = submit_transactions_to_mempool(smp, peer_id, request).await {
+        let smp_msg = MempoolSyncMsg {
+            message: Some(MempoolSyncMsg_oneof::SubmitTransactionsResponse(response)),
+        };
+        response_msg = Some(smp_msg);
+    }
+
+    if let Some(response_msg) = response_msg {
```

Comment:
looks like this is only executed if statement before is ok. Why not include it there then?
also we probably want to delivery error to caller even if we failed. It's better than timing out